src/main/kotlin/day18/repository/PhotoRepositoryImpl.kt

```
@@ -0,0 +1,24 @@
+
+class PhotoRepositoryImpl(private val dataSource: PhotoDataSource) : PhotoRepository {
+    override suspend fun getPhotos(query: String): Result<List<Photo>> {
+        return if (query.isBlank()) {
+            Result.Error(PhotoError.EmptyQuery)
+        } else {
+            try {
+                val photoDtos = dataSource.getPhotos(query)
+                val photos = photoDtos.toPhotoList()
+
+                Result.Success(photos)
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)